Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/WatchTab/WatchTabContent.tsx

```diff
@@ -0,0 +1,326 @@
+import { getMemoryVariables } from '../../../../../recoilModel/dispatchers/utils/project';
+import { WatchDataPayload } from '../../WatchVariableSelector/utils/helpers';
+
+import { WatchTabObjectValue } from './WatchTabObjectValue';
+
+const contentContainer = css`
+  display: flex;
+  flex-flow: column nowrap;
+  height: 100%;
+  width: 100%;
+`;
+
+const toolbar = css`
+  display: flex;
+  flex-flow: row nowrap;
+  flex-shrink: 0;
+  height: 24px;
+  padding: 8px 16px;
+`;
+
+const content = css`
+  display: flex;
+  flex-flow: column nowrap;
+  height: calc(100% - 40px);
+  width: 100%;
+`;
+
+const undefinedValue = css`
+  font-family: Segoe UI;
+  font-size: 12px;
+  font-style: italic;
+  height: 16px;
+  line-height: 16px;
+`;
+
+const watchTableStyles: Partial<IDetailsListStyles> = {
+  root: {
+    height: '100%',
+    selectors: {
+      '& > div[role="grid"]': {
+        height: '100%',
+      },
+    },
+  },
+  contentWrapper: {
+    overflowY: 'auto' as 'auto',
+    // fill remaining space after table header row
+    height: 'calc(100% - 60px)',
+  },
+};
+
+const rowStyles: Partial<IDetailsRowStyles> = {
+  cell: { minHeight: 32, padding: '8px 6px' },
+  checkCell: {
+    height: 32,
+    minHeight: 32,
+    selectors: {
+      '& > div[role="checkbox"]': {
+        height: 32,
+      },
+    },
+  },
+  root: { minHeight: 32 },
+};
+
+const addIcon: IIconProps = {
+  iconName: 'Add',
+};
+
+const removeIcon: IIconProps = {
+  iconName: 'Cancel',
+};
+
+const NameColumnKey = 'column1';
+const ValueColumnKey = 'column2';
+// TODO: update to office-ui-fabric-react@7.170.x to gain access to "flexGrow" column property to distribute proprotional column widths
+// (name column takes up 1/3 of space and value column takes up the remaining 2/3)
+const watchTableColumns: IColumn[] = [
+  {
+    key: NameColumnKey,
+    name: 'Name',
+    fieldName: 'name',
+    minWidth: 100,
+    maxWidth: 600,
+    isResizable: true,
+  },
+  {
+    key: ValueColumnKey,
+    name: 'Value',
+    fieldName: 'value',
+    minWidth: 100,
+    maxWidth: undefined,
+    isResizable: true,
+  },
+];
+
+const watchTableLayout: DetailsListLayoutMode = DetailsListLayoutMode.justified;
+
+// this can be exported and used in other places
+const getValueFromBotTraceScope = (delimitedProperty: string, botTrace: Activity) => {
+  const propertySegments = delimitedProperty.split('.');
+  const value = propertySegments.reduce(
+    (accumulator: object | string | number | boolean | undefined, segment, index) => {
+      // first try to grab the specified property off the root of the bot trace's memory
+      if (index === 0) {
+        return botTrace?.value[segment];
+      }
+      // if we are not on the root, try accessing the next value of the desired property
+      if (typeof accumulator === 'object') {
+        return accumulator[segment];
+      } else {
+        return undefined;
+      }
+    },
+    undefined
```

Comment:
?

Reply:
This is basically the equivalent of:

```
const segments = ['user.outer.inner.someValue'].split('.')
return botTrace?.value?.[segments[0]]?.[segments[1]]?.[segments[2]]?....[segments[n]]
```

Do you know of a better way to do this with a dynamic number of properties to try and access? This basically works its way from the root of the object down to the desired property.

Reply:
I suppose I could do something like:

```
let returnVal = botTrace?.value;
for const segment in segments.split('.') {
  returnVal = returnVal?.[segment]
}
return returnVal
```

Reply:
^^ Just tried this, it works and looks way cleaner.

Reply:
Sweet. I had a similar requirement as well. Used lodash get
https://github.com/microsoft/BotFramework-Composer/pull/8141/files#diff-c949aaa883b99af0c85135f197dfbdf385d73097d1f5ba2511fbe540a4f39a49R60.

Reply:
Awesome, I'll do that instead.